### ISSUE TITLE
Leave manifest platform empty for cross-platform bottles

### ISSUE THREAD

**Issue by justenstall** [Homebrew/brew#17395]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

The image index for cross-platform bottles lists a single manifest which is downloaded for all platforms. The manifest descriptor includes a "platform" object describing the architecture the bottle was built on. To conform to the OCI spec, the descriptor should exclude the "platform" field since it is not platform-specific.

From the [OCI Image Index Specification](https://github.com/opencontainers/image-spec/blob/main/image-index.md):

> - **`platform`** *object*
>
>     This OPTIONAL property describes the minimum runtime requirements of the image.
>     This property SHOULD be present if its target is platform-specific.

Example of a cross-platform bottle's image index: [`ghcr.io/homebrew/core/ca-certificates:2024-03-11`](https://github.com/Homebrew/brew/files/15505348/ca-certificates.index.json)

### What is the motivation for the feature?

This change will make bottles conform to the OCI specification better.

This change will also improve the inspection and discovery of bottles through the GitHub Packages UI and command line clients like `oras` and `skopeo`. The GitHub Package UI labels the images with the "platform" information, which is misleading:
![image](https://github.com/Homebrew/brew/assets/39888103/4a1c8dc1-06f1-45e1-b25d-8745d6bf80f5)

### How will the feature be relevant to at least 90% of Homebrew users?

This change will improve the bottle publishing format consumed by all Homebrew users.

### What alternatives to the feature have been considered?

None.

**Comment by MikeMcQuaid** [comment]
Makes sense, thanks @justenstall. For whoever picks this up: I note the tab also contains `arch: "arm64"` which should probably be removed and `built_on` contains the OS/OS version/Xcode version which could potentially be removed.

**Comment by MikeMcQuaid** [comment]
CC @Rylan12 in case this makes sense to do as part of the bulk updates

**Comment by MikeMcQuaid** [comment]
Passing on this. Will review a PR.
